stdlib/seq.mli

```diff
@@ -0,0 +1,57 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                 Simon Cruanes                                          *)
+(*                                                                        *)
+(*   Copyright 2017 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(* Module [Seq]: functional iterators *)
+
+(** {1 Functional Iterators} *)
+
+(** An iterator of type ['a Seq.t] represents a pure, lazy collection
+    of elements of type ['a]. Whatever its origin, an iterator can be
+    transformed or traversed in a uniform way. Iterators are
+    a convenient way to move data between collections, and
+    transforming them, without allocating a large list upfront.
+
+    {b Note}: programmers that expose iterators should take care that
+    the iterator can be traversed several times and should always yield
+    the same element. If this is impossible (e.g. when iterating over
+    lines on a socket) greater care must be taken, or {!memo} can be used
+    to save the elements when they are produced on the first traversal.
+*)
+
+(** @since NEXT_RELEASE *)
+
+type +'a node =
+  | Nil
+  | Cons of 'a * 'a t
+
+and 'a t = unit -> 'a node
+
+val empty : 'a t
+
+val return : 'a -> 'a t
+
+val map : ('a -> 'b) -> 'a t -> 'b t
+
+val filter : ('a -> bool) -> 'a t -> 'a t
+
+val filter_map : ('a -> 'b option) -> 'a t -> 'b t
+
+val flat_map : ('a -> 'b t) -> 'a t -> 'b t
+
+val fold_left : ('a -> 'b -> 'a) -> 'a -> 'b t -> 'a
+
+val iter : ('a -> unit) -> 'a t -> unit
+
+val memo : 'a t -> 'a t
```

Comment:
Hi, sorry for the late reply, it is more like [`next` in `Core.Sequence`](https://ocaml.janestreet.com/ocaml-core/latest/doc/core_kernel/Core_kernel/Sequence.mod/#/next.val). But we can wait until this PR is merged and then consider this function as a useful extension.  I personally used it a lot. Not sure how others feel.